Content.Shared/_Starlight/Antags/Abductor/AbductorsComponents.cs

```diff
@@ -0,0 +1,124 @@
+{
+    [DataField(readOnly: true)]
+    public EntProtoId? RemoteEntityProto = "AbductorHumanObservationConsoleEye";
+
+    [DataField, AutoNetworkedField]
+    public NetEntity? RemoteEntity;
+}
+[RegisterComponent, NetworkedComponent, Access(typeof(SharedAbductorSystem)), AutoGenerateComponentState]
+public sealed partial class AbductorConsoleComponent : Component
+{
+    [DataField, AutoNetworkedField]
+    public NetEntity? Target;
+
+    [DataField, AutoNetworkedField]
+    public NetEntity? AlienPod;
+
+    [DataField, AutoNetworkedField]
+    public NetEntity? Experimentator;
+}
+[RegisterComponent, NetworkedComponent, Access(typeof(SharedAbductorSystem))]
+public sealed partial class AbductorAlienPadComponent : Component
+{
+}
+[RegisterComponent, NetworkedComponent, Access(typeof(SharedAbductorSystem)), AutoGenerateComponentState]
+public sealed partial class AbductorExperimentatorComponent : Component
+{
+    [DataField, AutoNetworkedField]
+    public NetEntity? Console;
+
+    [DataField, ViewVariables(VVAccess.ReadWrite)]
+    public string ContainerId = "storage";
+}
+
+[RegisterComponent, NetworkedComponent, Access(typeof(SharedAbductorSystem)), AutoGenerateComponentState]
+public sealed partial class AbductorGizmoComponent : Component
+{
+    [DataField, AutoNetworkedField]
+    public NetEntity? Target;
+}
+
+[RegisterComponent, NetworkedComponent, Access(typeof(SharedAbductorSystem))]
+public sealed partial class AbductorComponent : Component
+{
+}
+
+[RegisterComponent, NetworkedComponent, AutoGenerateComponentState]
+public sealed partial class AbductorVictimComponent : Component
+{
+    [DataField("position"), AutoNetworkedField]
+    public EntityCoordinates? Position;
+
+    [DataField("organ"), AutoNetworkedField]
+    public AbductorOrganType Organ = AbductorOrganType.None;
+
+    [DataField]
+    public TimeSpan? LastActivation;
+}
+
+[RegisterComponent, NetworkedComponent, Access(typeof(SharedAbductorSystem)), AutoGenerateComponentState]
+public sealed partial class AbductorOrganComponent : Component
+{
+    [DataField("organ"), AutoNetworkedField]
+    public AbductorOrganType Organ;
+}
+
+[RegisterComponent, NetworkedComponent, Access(typeof(SharedAbductorSystem)), AutoGenerateComponentState]
+public sealed partial class AbductorScientistComponent : Component
+{
+    [DataField("position"), AutoNetworkedField]
+    public EntityCoordinates? SpawnPosition;
+}
+
+[RegisterComponent, NetworkedComponent, Access(typeof(SharedAbductorSystem)), AutoGenerateComponentState]
+public sealed partial class RemoteEyeSourceContainerComponent : Component
+{
+    [DataField, AutoNetworkedField]
+    public EntityUid? Actor;
+}
+
+[RegisterComponent, NetworkedComponent, Access(typeof(SharedAbductorSystem)), AutoGenerateComponentState]
+public sealed partial class AbductorsAbilitiesComponent : Component
+{
+    [DataField, AutoNetworkedField]
+    public EntityUid? ExitConsole;
```

Comment:
rather than storing `ExitConsole` and `SendYourself` as explicit things, it would be better to have a `NewAbilities` array instead